uid2_client/sharing_client.py

```diff
@@ -0,0 +1,78 @@
+"""Usage
+>>> from uid2_client import SharingClient
+"""
+import base64
+import datetime as dt
+
+from .client_type import ClientType
+from .encryption import encrypt, decrypt_token
+from .refresh_keys_util import refresh_sharing_keys
+
+
+class SharingClient:
+    """Client for interacting with UID2 Sharing services
+
+        You will need to have the base URL of the endpoint and a client key pair (auth/secret)
+        to consume web services.
+
+        Methods:
+            refresh_keys: Refresh encryption keys from UID2 servers
+            encrypt_raw_uid_into_sharing_token: encrypt a raw UID2 into a sharing token
+            decrypt_sharing_token_into_raw_uid: decrypt a sharing token
+        """
+
+    def __init__(self, base_url, auth_key, secret_key):
+        """Create a new SharingClient client.
+
+        Args:
+            base_url (str): base URL for all requests to UID2 services (e.g. 'https://prod.uidapi.com')
+            auth_key (str): authorization key for consuming the UID2 services
+            secret_key (str): secret key for consuming the UID2 services
+
+        Note:
+            Your authorization key will determine which UID2 services you are allowed to use.
+        """
+        self._keys = None
+        self._base_url = base_url
+        self._auth_key = auth_key
+        self._secret_key = base64.b64decode(secret_key)
+
+    def encrypt_raw_uid_into_sharing_token(self, uid2, keyset_id=None, now=dt.datetime.now(tz=dt.timezone.utc)):
```

Comment:
Ideally we don't want to have a `now` parameter on these public methods. In the .NET SDK the `DateTime` param is on the internal method.

Reply:
good catch. I missed that the C# method is internal. Changed the method signature